src/phases/encounter-phase.ts

```diff
@@ -154,6 +154,7 @@ export class EncounterPhase extends BattlePhase {
 
       this.scene.ui.setMode(Mode.MESSAGE).then(() => {
         if (!this.loaded) {
+          this.trySetWeatherIfNewBiome(); // Set weather before session gets saved
           //@ts-ignore
           this.scene.gameData.saveAll(this.scene, true, battle.waveIndex % 10 === 1 || this.scene.lastSavePlayTime >= 300).then(success => { // TODO: get rid of ts-ignore
```

Comment:
```suggestion
          this.scene.gameData.saveAll(this.scene, true, battle.waveIndex % 10 === 1 || (this.scene.lastSavePlayTime ?? 0) >= 300).then(success => {
```
Can you fix this while you're here?